### PR TITLE
Add algorithms_random_node.java in lists section

### PR DESCRIPTION
**Pull request by skmodi649**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All functions and variable names follow Java naming conventions.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{2850}`.

### CODE REVIEW

**Review by skmodi649 — COMMENTED** [review]
Reviewed!

**Review by skmodi649 — COMMENTED** [review]
Prettified the code!

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * Step 7 : STOP
+ */
+
+
+package com.thealgorithms.datastructures.lists;
+
+
+import java.util.ArrayList;
+
+public class Algorithms_random_node {
```
Comment (yanglbme) [inline src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java:31]: Class Name should be `Camel-Case`

Comment (yanglbme) [inline src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java:31]:
```suggestion
public class RandomNode {
```

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    static class ListNode {
+        int val;
+        ListNode next;
+
+        ListNode(int val) {
+            this.val = val;
+        }
+    }
+    public Algorithms_random_node(ListNode head) {
+        list = new ArrayList < > ();
```
Comment (yanglbme) [inline src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java:47]:
```suggestion
        list = new ArrayList <>();
```

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        head = new ListNode(15);
+        head.next = new ListNode(25);
+        head.next.next = new ListNode(4);
+        head.next.next.next = new ListNode(1);
+        head.next.next.next.next = new ListNode(78);
+        head.next.next.next.next.next = new ListNode(63);
+        Algorithms_random_node list = new Algorithms_random_node(head);
+
+        int random_num = list.getRandom();
```
Comment (yanglbme) [inline src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java:79]:
```suggestion
        int randomNum = list.getRandom();
```

**Review by yanglbme — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java (hunk 1 above)
Comment (skmodi649) [reply src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java]: Change made!

Comment (skmodi649) [reply src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java]: Change made!

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java (hunk 2 above)
Comment (skmodi649) [reply src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java]: Change made!

File: src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java (hunk 3 above)
Comment (skmodi649) [reply src/main/java/com/thealgorithms/datastructures/lists/Algorithms_random_node.java]: Change made!

**Review by skmodi649 — COMMENTED** [review]
Changes made!

**Review by skmodi649 — COMMENTED** [review]
Changes made and reviewed!

**Review by yanglbme — APPROVED** [review]

**Review by skmodi649 — COMMENTED** [review]
Reviewed!

**Pull request merged**